recipes/libcublas/meta.yaml

```diff
@@ -0,0 +1,103 @@
+{% set name = "libcublas" %}
+{% set version = "12.0.1.189" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 38272cc9184b4a57a6250fc2cd914ffc827f9f8cd5a3e1356bc3c31f843caf8f  # [linux64]
+  sha256: 2d46e735f8e76e2eb7df1b4f41d56aa8d5d2683f99988bac82d6ae33fb6d344b  # [ppc64le]
+  sha256: 96b68f6821684f39f338862639ce74a74c75830bf44dcd17ef372a1cb8c2304d  # [aarch64]
+  sha256: 88f29bec81880844da2eee13f1cdc51e1ccd8b4c60af6df633577810b30f80d0  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+  run_exports:
+    strong:
+      - __glibc >=2.17
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  host:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+    - cuda-version 12.0
+  run:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+    - {{ pin_compatible("cuda-version") }}
```

Comment:
Do some of these host/run pinnings need to be moved to the individual packages, rather than the recipe as a whole? I can't remember the rules for how split packages propagate dependencies, but I think the `run` dependencies are supposed to be defined in the subpackages.

Reply:
The package artifacts are made available on CI

Though it seems this PR preceded that change. Have merged in the latest changes so we can get a fresh build with those packages uploaded

For example taking [this build]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=673765&view=results ), we can click the "N published" link 

<details>

<img width="1840" alt="Screen Shot 2023-03-15 at 2 05 49 PM" src="https://user-images.githubusercontent.com/3019665/225442651-9d4449cf-32c7-4244-8828-45d81085ce1e.png">

</details>

We can then select the build archive to download

<details>

<img width="1840" alt="Screen Shot 2023-03-15 at 2 09 22 PM" src="https://user-images.githubusercontent.com/3019665/225443681-0335c530-250e-46e7-838b-4aba71218d5c.png">

</details>

The decompressed archive then contains the Conda package within. If one is logged into Azure, the `.conda` package can be downloaded directly from Azure.

<details>

```bash
$ tree conda_pkgs_noarch
conda_pkgs_noarch
├── current_repodata.json
├── current_repodata.json.bz2
├── index.html
├── python-lsp-ruff-1.2.0-pyhd8ed1ab_0.conda
├── repodata.json
├── repodata.json.bz2
├── repodata_from_packages.json
└── repodata_from_packages.json.bz2
```

</details>

The `.conda` package is just a `zip` file (and the extension can be changed). Within it contains `zst` files, which can be explored with any tool that understands that format (like [PeaZip]( https://peazip.github.io ))

Reply:
Cool so now stage-recipes by default retains the artifacts?

Reply:
Yeah though I think they expire at some point

FWICT `libcublas` depends on `cuda-version`, but this doesn't happen for the other packages. Should add this also affects other things (like where compiler runtime libraries are included)

So think we should move `cuda-version` to the dependencies

A separate question is whether we want to do this with `sysroot` & compiler runtime libraries

Reply:
> So think we should move `cuda-version` to the dependencies

As in move it to the sub-package? Or what do you mean with "move to the dependencies", for me `requirements.run` is (one of) "the dependencies" :D

Reply:
Yes, move `run` dependencies to subpackages. Those don't propagate from the top-level afaik. Examples: 

https://github.com/rapidsai/cudf/blob/8fbfb4ae5f083ac8473023f9e4982e8021b83c6d/conda/recipes/libcudf/meta.yaml#L34
https://github.com/rapidsai/cudf/blob/8fbfb4ae5f083ac8473023f9e4982e8021b83c6d/conda/recipes/libcudf/meta.yaml#L68

Reply:
Have tried to suggest some changes below to clear this up